Celeste.Mod.mm/Patches/LevelLoader.cs

```diff
@@ -155,8 +157,19 @@ public patch_LevelLoader(Session session, Vector2? startPosition = default)
                 GFX.PortraitsSpriteBank = new SpriteBank(GFX.Portraits, path);
             } catch (Exception e) {
                 string sid = session?.Area.GetSID() ?? "NULL";
-                patch_LevelEnter.ErrorMessage = Dialog.Get("postcard_levelloadfailed").Replace("((sid))", sid);
-                Logger.Log(LogLevel.Warn, "LevelLoader", $"Failed loading map {sid}");
+                if (patch_LevelEnter.ErrorMessage == null) {
+                    if (e is XmlException) {
+                        string lineNum = Regex.Match(e.Message, @"Line (\d+)").Groups[1].Value;
+                        patch_LevelEnter.ErrorMessage = Dialog.Get("postcard_xmlerror").Replace("((path))", path).Replace("((line))", lineNum);
+                        Logger.Log(LogLevel.Warn, "LevelLoader", $"Failed parsing {path}");
+                    } else if (e.StackTrace.Contains("Autotiler")) {
```

Comment:
Is this really the cleanest way to determine that the error comes from the Autotiler? Could you use something like `e.TargetSite`, or do these errors generally happen in a called method that's no longer in `Autotiler`?

Reply:
I'll look into `TargetSite`, pretty much every other `Exception` "feature" is just user-defined data so I guess I skipped over that. Will try and get rid of all the `StackTrace` string checks too